CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/rest/RestClientNGTest.java

```diff
@@ -0,0 +1,412 @@
+    
+    public RestClientNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of generateUrl method, of class RestClient.
+     */
+    @Test
+    public void testGenerateUrl_NullParams() throws Exception {
+        System.out.println("testGenerateUrl");
+        String url = "https://testurl.tst/testEndpoint";
+        List<Tuple<String, String>> params = null;
+        URL result = RestClient.generateUrl(url, params);
+        assertEquals(result, new URL(url));
+    }
+
+    /**
+     * Test of generateUrl method, of class RestClient.
+     * @throws java.lang.Exception
+     */
+    @Test
+    public void testGenerateUrl_Empty() throws Exception {
+        System.out.println("testGenerateUrl");
+        String url = "https://testurl.tst/testEndpoint";
+        List<Tuple<String, String>> params = new ArrayList<>();
+        URL result = RestClient.generateUrl(url, params);
+        assertEquals(result, new URL(url));
+    }
+
+    /**
+     * Test of generateUrl method, of class RestClient.
+     */
+    @Test
+    public void testGenerateUrl_InvalidParamName() throws Exception {
+        System.out.println("testGenerateUrl");
+        String url = "https://testurl.tst/testEndpoint";
+        List<Tuple<String, String>> params = new ArrayList<>();
+        params.add(new Tuple("", "value"));
+        URL result = RestClient.generateUrl(url, params);
+        assertEquals(result, new URL(url));
+    }
+
+    /**
+     * Test of generateUrl method, of class RestClient.
+     */
+    @Test
+    public void testGenerateUrl_ValidParams() throws Exception {
+        System.out.println("testGenerateUrl");
+        String url = "https://testurl.tst/testEndpoint";
+        List<Tuple<String, String>> params = new ArrayList<>();
+        params.add(new Tuple("param1", "value1"));
+        params.add(new Tuple("param2", "value2 with spaces"));
+        URL result = RestClient.generateUrl(url, params);
+        assertEquals(result, new URL(String.format("%s?param1=value1&param2=value2+with+spaces", url)));
+        result = RestClient.generateUrl(url, params, "%20");
+        assertEquals(result, new URL(String.format("%s?param1=value1&param2=value2%%20with%%20spaces", url)));
+    }
+
+    /**
+     * Test of beforeGet method, of class RestClient.
+     */
+    @Test
+    public void testBeforeGet() {
+        System.out.println("testBeforeGet");
+        String url = "";
+        List<Tuple<String, String>> params = null;
+        RestClient instance = new RestClientImpl(0, "", null);
+        instance.beforeGet(url, params);
+        // Method does nothing - designed to be overridden, test really just
+        // checks it can be called
```

Comment:
Personally I would only bother adding tests for functions that have anything to be tested. Tests for empty functions don't really add anything to code coverage and just blow out the unit test class more than it need to. In reality, any classes overriding this function should have unit tests for the implementation...but that doesn't need to be dealt with here

Reply:
left in as they do add coverage, although as mentioned they don't test 'muxh'